### PR TITLE
Ticket4659 fix git support

### PR DESCRIPTION
**Pull request by NikolaRoev**
https://github.com/ISISComputingGroup/IBEX/issues/4659

### CODE REVIEW

File: utils/requests_utils.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        },
+        json={
+            "name": name,
+            "visibility": "public",
+            "auto_init": True
+        }
+    )
+
+    if response.status_code == requests.codes["created"]:
+        logging.info("Successfully created GitHub repository.")
```
Comment (rerpha) [inline utils/requests_utils.py]: wonder if we could logging.info() the github URL here? looks like it's part of the response JSON under `html_url`?

File: IBEX_device_generator.py [hunk 2]
```diff
@@ -57,11 +61,11 @@ def main():
     parser.add_argument("--name", type=str, help="Name of the device", required=True)
     parser.add_argument("--ticket", type=int, help="Ticket number", required=True)
     parser.add_argument("--device_count", type=int, help="Number of duplicate IOCs to generate", default=2)
-    parser.add_argument("--use_git", action='store_true', help="Currently not working - Use to create relevant "
-                                                               "branches and support repository")
+    parser.add_argument("--use_git", action='store_true', help="Use to create relevant branches and support repository")
```
Comment (rerpha) [inline IBEX_device_generator.py]: I think we should now make this default to true (by using store_false?) and make the --token argument required if use_git is true as it now works

File: utils/requests_utils.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        use_git: Use Git, if True try to create the GitHub repository, else continue.
+        token: The GitHub authentication token. If a token was not given the script exits.
+        name: The name of the GitHub repository.
+    """
+    if not use_git:
+        return
+
+    if token is None:
+        logging.critical("If you are using Git, you need to add a GitHub token.")
+        exit()
```
Comment (rerpha) [inline utils/requests_utils.py]: we might be able to remove this if we make it a required arg if use_git is true

File: utils/git_utils.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ def __init__(self, path):
             self._repo = Repo(path)
         except (InvalidGitRepositoryError, NoSuchPathError):
             mkdir(path)
-            self._repo = Repo.init(path)
+            self._repo = Repo.init(path, kwargs="--initial-branch=main")
```
Comment (FreddieAkeroyd) [inline utils/git_utils.py]: Does this work ok? `kwargs` is normally pseudo syntax for a list of keyword arguments, https://stackoverflow.com/questions/64470052/how-to-set-default-branch-for-gitpython would suggest that `initial_branch='main'` would work

**Comment by FreddieAkeroyd** [comment]
The previous workflow of creating an empty GitHub repository by hand and then the generator script using you local cached git credentials for command line tools to create branches etc. is no longer supported. Should this still be allowed if `github_token` is not specified?

**Comment by rerpha** [comment]
with use_git but no token the code (correctly) doesn't create the remote repo, but still prints out `Attempting to create repository using remote {}. Press return to confirm it exists` - we should stop it from printing this as it implies that the remote URL has been created

**Pull request merged**